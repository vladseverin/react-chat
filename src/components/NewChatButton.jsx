import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import TextField from '@material-ui/core/TextField';
import red from '@material-ui/core/colors/red';

const styles = theme => ({
  newChatButton: {
    position: 'absolute',
    left: 'auto',
    right: theme.spacing.unit * 3,
    // eslint-disable-next-line
    bottom: theme.spacing.unit * 3 + 48,
  },
  modalWrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modal: {
    outlineColor: red[500],
    width: '30%',
    minWidth: '300px',
    padding: theme.spacing.unit * 3,
  },
});

class NewChatButton extends React.Component {
  static propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
  };

  state = {
    open: false,
    title: {
      value: '',
      isValid: true,
    },
  };

  toggleModal = () => {
    this.setState({ open: !this.state.open });
  };

  handleTitleChange = (event) => {
    this.setState({
      title: {
        value: event.target.value,
        isValid: true,
      },
    });
  };

  handleCreateClick = (event) => {
    event.preventDefault();

    const { title } = this.state;

    if (!title.value) {
      this.setState({
        title: {
          value: title.value,
          isValid: false,
        },
      });

      return;
    }

    this.props.onClick(title.value);
    this.toggleModal();
    this.setState({
      title: {
        value: '',
        isValid: true,
      },
    });
  };

  render() {
    const { classes, disabled } = this.props;
    const { open, title } = this.state;

    return (
      <React.Fragment>
        <Button
          variant="fab"
          disabled={disabled}
          color="primary"
          aria-label="add"
          className={classes.newChatButton}
          onClick={this.toggleModal}
        >
          <AddIcon />
        </Button>

        <Modal open={open} className={classes.modalWrapper} onClose={this.toggleModal}>
          <Paper className={classes.modal}>
            <Typography variant="title" id="modal-title">
              Create new chat
            </Typography>
            <TextField
              required
              fullWidth
              label="New chat"
              placeholder="Type the title..."
              type="text"
              margin="normal"
              autoComplete="new-chat"
              value={title.value}
              onChange={this.handleTitleChange}
              error={!title.isValid}
            />
            <Button color="primary" onClick={this.handleCreateClick}>
              Create
            </Button>
          </Paper>
        </Modal>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(NewChatButton);
