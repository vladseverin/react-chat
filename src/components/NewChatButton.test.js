import React from 'react';
import ReactDOM from 'react-dom';
import NewChatButton from './NewChatButton';

const mockProps = {
  onClick: jest.fn(),
  disabled: false,
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NewChatButton {...mockProps} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
