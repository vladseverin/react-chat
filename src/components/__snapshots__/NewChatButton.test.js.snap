// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewChatButton /> renders correctly 1`] = `
<button
  aria-label="add"
  className="MuiButtonBase-root-57 MuiButton-root-41 MuiButton-raised-47 MuiButton-fab-52 MuiButton-raisedPrimary-48 NewChatButton-newChatButton-38"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-42"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-60"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-66"
  />
</button>
`;
